dart.yaml

```diff
@@ -0,0 +1,134 @@
+package:
+  name: dart
+  version: 3.5.3
+  epoch: 0
+  description: The Dart SDK, including the VM, JS and Wasm compilers, analysis, core libraries, and more.
+  copyright:
+    - license: BSD-3-Clause
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
```

Comment:
_NOTE: info not a blocker for this package._ 

In dart ecosystem, it's very [common](https://github.com/search?q=%22COPY+--from%3Dbuild+%2Fruntime%2F+%2F%22+AND+%22dart%22&type=code) to use the following. Many template generators generate this including the official one. 

following is the second stage of the build pipeline:
```docker
FROM scratch AS server
COPY --from=build /runtime/ /
```

Now, as long as `scratch` is used, it's fine here. If any image is used that already ships with existing ca-certificates like wolfi, alpine etc. then the following error will occur.

```bash
 => ERROR [stage-1 2/3] COPY --from=build /runtime/ /                                                                                                                                    0.0s
------
 > [stage-1 2/3] COPY --from=build /runtime/ /:
------
Dockerfile:16
--------------------
  14 |     # and the pre-built AOT-runtime in the `/runtime/` directory of the base image.
  15 |     FROM alpine
  16 | >>> COPY --from=build /runtime/ /
  17 |     COPY --from=build /app/bin/server /app/bin/
  18 |
--------------------
ERROR: failed to solve: cannot replace to directory /var/lib/docker/fuse-overlayfs/yy0qk73q5lwnk1o8hmm1856pc/merged/etc/ssl/certs with file
```

I don't think, as per my understanding currently, there's way we can avoid this, ca-certificates-bundle is a runtime dependency for dart.
so far the image is working good and can used directly as long as `scratch` is used in the second stage.

Reply:
ouch.

I almost want to make "/runtime" a symlink to "/" or "/usr"

Can you point at sample, demo apps that do that?

Cause if we need to revendor runtime like this to be self-contained, we would also need re-vendor relevant json sbom. And apk  database will look odd.

Reply:
hang on, in a distroless world, our dart image would be minimal. It can contain /runtime symlink to either / or /usr, would that be enough?

Cause I want to avoid building nested containers.

Reply:
I agree and like the idea, I'm trying to think and cover the case where people are already using a multistage pipeline and given upstream vendors libraries under `/runtime` and users are copying that runtime inside scratch image using `COPY --from=build /runtime/ /` 

ref: https://github.com/dart-lang/dart-docker/blob/a8ada07907fbfa139635fe90bfb31a236b7c2f34/Dockerfile-debian.template#L36-L51

If we don't ship the libraries and just copy the binary then existing Dockerfile won't work. From distroless world, if we use any of `scratch` or static images from cgr or gcr it won't work because underlying libraries are missing in these images. 

the binary is dynamically linking to some system libraries.
A simple binary compiled using binary links to the following:
```bash
 $ dart compile exe bin/server.dart  -o bin/server
Generated: /home/user/examples/webserver/bin/server
$ ldd bin/server
        linux-vdso.so.1 (0x00007ffe7a520000)
        libdl.so.2 => /lib/libdl.so.2 (0x00007cbcfa090000)
        libpthread.so.0 => /lib/libpthread.so.0 (0x00007cbcfa08b000)
        libm.so.6 => /lib/libm.so.6 (0x00007cbcf9fa9000)
        libc.so.6 => /lib/libc.so.6 (0x00007cbcf980b000)
        /lib64/ld-linux-x86-64.so.2 (0x00007cbcfa098000)
```

Usage on GitHub shows that this image is used in multistage pipeline so we'll either need a `-compat` vendoring libraries (hack) or a direct copy of that at runtime compatibility with existing distroless usage. 

I've reverted symlinking for now locally but open to more discussion here on the best way to proceed here. We can ship with our best practices and write a migration note if required.

Reply:
I think i replied in the wrong thread https://github.com/wolfi-dev/os/pull/29644#discussion_r1787707357